databricks/koalas/indexing.py

```diff
@@ -75,12 +75,169 @@ def _unfold(key, col):
     return rows_sel, cols_sel
 
 
-class SparkDataFrameLocator(object):
+class LocIndexer(object):
     """
-    A locator to slice a group of rows and columns by conditional and label(s).
+    Access a group of rows and columns by label(s) or a boolean Series.
 
-    Allowed inputs are a slice with all indices or conditional for rows, and string(s) or
-    :class:`Column`(s) for cols.
+    ``.loc[]`` is primarily label based, but may also be used with a
+    conditional boolean Series derived from the DataFrame or Series.
+
+    Allowed inputs are:
+
+    - A single label, e.g. ``5`` or ``'a'``, (note that ``5`` is
+      interpreted as a *label* of the index, and **never** as an
+      integer position along the index) for column selection.
+
+      .. warning:: A single label for row selection is not allowed
+
+    - A list or array of labels, e.g. ``['a', 'b', 'c']``.
+
+      .. warning:: With a list or array of labels for row selection,
+          Koalas does not preserve the given order but preserves the data order.
```

Comment:
i don't understand what given order vs data order means here.

Reply:
hm, how can I explain ..

Let's see pandas' behavior:

```py
>>> import pandas as pd
>>> pdf = pd.DataFrame([[1, 2], [4, 5], [7, 8]], index=['cobra', 'viper', 'sidewinder'], columns=['max_speed', 'shield'])
>>> pdf.loc[['sidewinder', 'viper']]
            max_speed  shield
sidewinder          7       8
viper               4       5
>>> pdf.loc[['viper', 'sidewinder']]
            max_speed  shield
viper               4       5
sidewinder          7       8
```

The results were reordered by the specified column name order, but Koalas behaves just a filter without reordering.